src/main/java/org/spongepowered/common/mixin/core/item/MixinItemFishingRod.java

```diff
@@ -54,10 +54,12 @@ public ItemStack onItemRightClick(ItemStack itemStack, World world, EntityPlayer
             player.fishEntity.handleHookRetraction();
         } else {
             EntityFishHook fishHook = new EntityFishHook(world, player);
-            if (!Sponge.getGame().getEventManager()
+            if (world.isRemote || !Sponge.getGame().getEventManager()
```

Comment:
mmmm good point @JBYoshi . While the property in the annotation can be optional, the value must be set in the annotation's class itself. DEFAULT could be added to Type in platform (but only really makes sense in events).

Type.DEFAULT - External code doesn't care about the side, can do it on either/or.
Type.SERVER - External code only wants to execute on server thread
Type.CLIENT - External code only wants to execute on client thread

This is what it would mean for the @Subscribe annotation.